bots/code_review_master.rb

```diff
@@ -0,0 +1,202 @@
+          end
+        end
+
+        ###################
+        # Clear reviewers #
+        ###################
+        if clear_reviewers_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is clearing all reviewers"
+          )
+          if data.reviewers.any? && data.clear_reviewers!
+            m.send_message("Reviewers have been cleared!")
+          else
+            m.send_message("No reviewers yet! Add one like `codereviewmaster add <name>`")
+          end
+        end
+
+        ####################
+        # Output help text #
+        ####################
+        if help_output_request?(message)
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is requesting help output"
+          )
+          help = <<-HELP_TEXT
+Hi! I help assign code reviewers.
+
+*Assign a reviewer*
+
+`codereviewmaster g/123456`
+
+Assigns a code reviewer to the provided gerrit change id. Posts in slack &
+assigns a reviewer in gerrit (assuming gerrit is configured).
+
+*List reviewers*
+
+`codereviewmaster list`
+
+Shows list of existing reviewers.
+
+*Add a reviewer*
+
+`codereviewmaster add john`
+
+or
+
+`codereviewmaster add slack:john gerrit:jcorrigan`
+
+Adds a new reviewer. Dups are ignored. If you just provide a string
+it assumes the same name for both slack & gerrit. If they are different,
+use the second syntax.
+
+*Remove a reviewer*
+
+`codereviewmaster remove jcorrigan`
+
+Works with either username.
+
+*Clear all reviewers*
+
+`codereviewmaster clear`
+
+Removes all reviewers
+
+*Help output*
+
+`codereviewmaster help`
+
+Outputs this text
+          HELP_TEXT
+          m.send_message(help)
+        end
+
+        #####################
+        # Assign a reviewer #
+        #####################
+        if (change_id = assign_review_request?(message))
+          SlackbotFrd::Log.debug(
+            "user '#{user}' in channel '#{channel}' is requesting a code review " \
+            "for change_id: #{change_id}"
+          )
+          if data.reviewers.any?
+            assigned = data.assign_reviewer!(change_id)
+            SlackbotFrd::Log.debug(
+              "user '#{assigned}' chosen to review '#{change_id}'"
+            )
+            m.send_message("Hey #{assigned} :wave-1:, can you review this :point_up:")
+
+            unless data.persist_to_gerrit!(
+              change_id: change_id,
+              assigned: assigned
+            )
+              m.send_message("Hm, I was unable to add #{assigned} as a reviewer in gerrit.")
+            end
+          else
+            m.send_message("No reviewers yet! Add one like `codereviewmaster add <name>`")
+          end
+        end
+      end
+    end
+  end
+
+  private
+
+  ######################
+  # Determine requests #
+  ######################
+  def add_reviewer_request?(message)
+    (message.downcase.match(/^#{CODEREVIEWMASTER} #{ADD_COMMAND} (.+)/) || [])[1]
```

Comment:
You can just use the `i` option in the regex match for case insensitivity so you don't have to `downcase` the message every time.

Reply:
👍